### ISSUE TITLE
Can no longer add roles to service accounts from apb service account. 

### ISSUE THREAD

**Issue by maleck13** [openshift/ansible-service-broker#399]
**What happened**:
When running a provision of an existing apb, some of the commands started to fail.
The specific command was to add the edit role to a service account that is used to run the application.
**What you expected to happen**:
Continue be able to add a role in a namespace to the given service account.
**How to reproduce it**:
In an apb create a service account and then run
```
oc policy add-role-to-user edit system:serviceaccount:'{{namespace}}':'{{serviceAccount}}' -n '{{namespace}}'
```

**Result**
```
Error from server (Forbidden): User \"system:serviceaccount:myproject:apb-1b9524ae-ee29-4cea-85eb-aca2454c71e9\" cannot list rolebindings in project \"myproject\""
```

For reference this issue is discussed in the PR also:
https://github.com/openshift/ansible-service-broker/pull/393

**Comment by maleck13** [comment]
To give more detail on the use case. Might help with framing the issue. 

We want to provide dynamic configuration for external mobile clients to service bindings within a given namespace.
To achieve this, these client will call  to the mobile server at startup with an API Key. The server then uses its service account to read the binding secrets and create a configuration object to send back to the client. 
This allows for services such as remote debugging to be enabled and disabled from within the mobile server ui.

**Comment by djzager** [comment]
Hey @maleck13 just wanted to make you aware that what I am working to do is make the `role` that we give to the apb configurable via the broker's configuration file. I'll reference this issue when I put up the PR.

**Comment by maleck13** [comment]
@djzager Thanks for the update. One concern I have here, is that if we need to use this option to upgrade the permissions, it makes us "non standard" it is important to us that we users are able to install the mobile server from the catalog both for openshift online, on prem and dedidicated.

**Comment by djzager** [comment]
@maleck13

> Thanks for the update. One concern I have here, is that if we need to use this option to upgrade the permissions, it makes us "non standard" it is important to us that users are able to install the mobile server from the catalog both for openshift online, on prem and dedidicated.

I understand the concern. I don't believe that our intention is to keep you in a non-standard setup, but rather allow you to continue to work with the latest broker while we are still figuring out how we will secure the broker while allowing APBs like yours to function as expected. As I understand the end goal, we would like to have APBs requesting what they need access to (including roles). This way, cluster administrators will have power over who can execute what APBs and what those APBs are allowed to do. 

> Would definitely like to understand more why applying a role to a service account within a namespace is something that we don't want to do from within an apb.

I think the issue is more like this, most APBs do not need the ability to apply roles to service accounts, and for this reason we elected to make the default role for the APB sandbox `edit`. This in no way means that what you are trying to accomplish is not a valid use-case that we should support. Keeping in mind the long-term goal of having APBs make it known what permissions they need in order to run successfully, then there is this future where APB developers have the power to do what they need to do and cluster admins have the power to not let certain APBs to execute.

**Comment by maleck13** [comment]
@djzager Great ok thanks for the update.

**Comment by jwmatthews** [comment]
I'm reopening the issue so we can track a standard path for allowing an app like Keycloack which may require a role that is not configured by default to run.

**Comment by maleck13** [comment]
@jwmatthews Just noticed you called out keycloak there. Just to be clear, the apb that we are developing that needs to use a service account it the mobile control panel. 
https://github.com/feedhenry/mcp-standalone/tree/master/cmd/mcp-apb

**Comment by jmrodri** [comment]
Fixed by PR #403
